learn/tooling-guide/visual-studio-code-extension/language-support/language-support.md

```diff
@@ -2,16 +2,127 @@
+active: intellisense
+intro: The Visual Studio Code Ballerina extension brings in language support to enhance the development experience and increase its efficiency. This is built into the extension via a Language Server implementation, which consists of the below features.
+description: The Visual Studio Code Ballerina extension brings in language support to enhance the development experience and increase its efficiency. This is built into the extension via a Language Server implementation, which consists of the below features.
 redirect_from:
-  - /learn/tooling-guide/vs-code-extension/language-support
-  - /learn/tooling-guide/vs-code-extension/language-support/
+  - /learn/tools-ides/vscode-plugin/language-intelligence
+  - /learn/tools-ides/vscode-plugin/language-intelligence/
+  - /learn/vscode-plugin/language-intelligence/
+  - /learn/vscode-plugin/language-intelligence
+  - /learn/tools-ides/setting-up-visual-studio-code/language-intelligence
+  - /learn/tools-ides/setting-up-visual-studio-code/language-intelligence/
+  - /learn/setting-up-visual-studio-code/language-intelligence
+  - /learn/setting-up-visual-studio-code/language-intelligence/
+  - /learn/language-intelligence
+  - /learn/language-intelligence/
+  - /swan-lake/learn/getting-started/setting-up-visual-studio-code/language-intelligence/
+  - /swan-lake/learn/getting-started/setting-up-visual-studio-code/language-intelligence
+  - /learn/tooling-guide/vs-code-extension/language-support/language-intelligence
+  - /learn/tooling-guide/vs-code-extension/language-support/language-intelligence/
+  - /learn/tooling-guide/visual-studio-code-extension/language-support/language-intelligence
+  - /learn/tooling-guide/visual-studio-code-extension/language-support/language-intelligence/
   - /learn/tooling-guide/visual-studio-code-extension/language-support
 ---
 
-## Language Intelligence
+## IntelliSense
+
+The below are the features related to IntelliSense.
+
+### Code Completion
+
+The extension provides you with suggestions on variables, keywords, and code snippets of language constructs (such as functions, type definitions, services, iterable constructs, etc.)
+
+![Code Completion](/learn/images/code-completion.gif)
+
+#### Symbol Information on Hover
+
+When hovering over a symbol name, you will be provided with quick information about the particular symbol. For example, when hovering over a function name, you will be prompted with the associated documentation.
+
+![Symbol Information on Hover](/learn/images/symbol-information-on-hover.gif)
+
+#### Signature Help
+
+When typing a function/method call expression, the signature help will show information such as the function/method call’s description and parameter information. Signature help will be triggered when typing the open parenthesis and comma.
+
+![Signature Help](/learn/images/signature-help.gif)
+
+## Code Navigation
+
+The below are the features related to code navigation.
```

Comment:
Do we need this line?

Reply:
yes, better to have a description for the title.

Reply:
Removed.